### PR TITLE
fix: make workspace restart more informative

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
- bumps to `@eclipse-che/che-devworkspace-generator:7.86.0`
- displays a reason of a failure when trying to restart the workspace from a local devfile
- allows to easily open the devfile by a notification button

Screenshot from 2024-05-28 18-55-59

### What issues does this PR fix?

https://github.com/eclipse-che/che/issues/22913

### How to test this PR?
- create a repository with https://github.com/vitaliy-guliy/vscode-test-extension?editor-image=quay.io/che-incubator-pull-requests/che-code:pr-363-amd64
- install yaml extension
- open `.devfile.yaml` and make some invalid change. I usually add the container environment variable
```
      env:
        - name: test
          value: true
```
The `yaml` extension should underline `true` as one that has wrong type (considering devfile schema it should be string instead of boolean).

- run `Eclipse Che: Restart Workspace from Local Devfile` command and select the devfile
( if there is only one devfile in the project, you can press `Enter` several times to confirm the reload )

- at this step you should be warned with an explanation of the reason of failure

### Does this PR contain changes that override default upstream Code-OSS behavior?
- [ ] the PR contains changes in the [code](https://github.com/che-incubator/che-code/tree/main/code) folder (you can skip it if your changes are placed in a che extension )
- [ ] the corresponding items were added to the [CHANGELOG.md](https://github.com/che-incubator/che-code/blob/main/.rebase/CHANGELOG.md) file
- [ ] rules for automatic `git rebase` were added to the [.rebase](https://github.com/che-incubator/che-code/tree/main/.rebase) folder

### CODE REVIEW

File: code/extensions/che-remote/package.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
     "@types/fs-extra": "^9.0.13",
     "@types/jest": "^27.4.0",
     "@types/js-yaml": "^4.0.5",
-    "@types/node": "18.x"
+    "@types/node": "18.x",
+    "eslint": "^8.0.0"
   },
```
Comment (RomanNikitenko) [inline code/extensions/che-remote/package.json:46]: please update yarn.lock file

Comment (vitaliy-guliy) [reply code/extensions/che-remote/package.json:46]: done

**Review by RomanNikitenko — APPROVED** [review]
tested, 
works as described in the PR description

**Review by svor — APPROVED** [review]
lgtm

screenshot-che-dogfooding apps che-dev x6e0 p1 openshiftapps com-2024 05 31-13_07_55

**Comment by the-it-jaeger** [comment]
This change is great @vitaliy-guliy, thank you! 

I found one edge case: If the ServiceAccount running the DevWorkspace Pod does not have the appropriate k8s Role (permissions), you still get ambiguous HTTP Request errors when trying to perform actions such as "Dev Spaces: Restart Workspace from Local Devfile."

I discovered this when I had to override the default ServiceAccount so I could apply a custom SecurityContextContstraing and some extra securityContext for a special container component that had to run as a non-random user.

**Pull request merged**